### PR TITLE
feat: WorkspaceResources interface and List method

### PR DESCRIPTION
**Pull request by stefan-kiss**
## Description

Implementing missing Workspace Resources api 

## Testing plan
-

## External links

- [workspace-resources](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspace-resources)
- [Open Issue #754](https://github.com/hashicorp/go-tfe/issues/754)

## Output from tests

```
go test ./... -run TestWorkspaceResourcesList -timeout=30m
?   	github.com/hashicorp/go-tfe/examples/backing_data	[no test files]
?   	github.com/hashicorp/go-tfe/examples/configuration_versions	[no test files]
?   	github.com/hashicorp/go-tfe/examples/organizations	[no test files]
?   	github.com/hashicorp/go-tfe/examples/registry_modules	[no test files]
?   	github.com/hashicorp/go-tfe/examples/run_errors	[no test files]
?   	github.com/hashicorp/go-tfe/examples/state_versions	[no test files]
?   	github.com/hashicorp/go-tfe/examples/users	[no test files]
?   	github.com/hashicorp/go-tfe/examples/workspaces	[no test files]
?   	github.com/hashicorp/go-tfe/mocks	[no test files]
ok  	github.com/hashicorp/go-tfe	7.276s
```

### CODE REVIEW

File: workspace_resources.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	ID                       string `jsonapi:"primary,resources"`
+	Address                  string `jsonapi:"attr,address"`
+	Name                     string `jsonapi:"attr,name"`
+	CreatedAt                string `jsonapi:"attr,created-at"`
+	UpdatedAt                string `jsonapi:"attr,updated-at"`
+	Module                   string `jsonapi:"attr,module"`
+	Provider                 string `jsonapi:"attr,provider"`
+	ProviderType             string `jsonapi:"attr,provider-type"`
+	ModifiedByStateVersionID string `jsonapi:"attr,modified-by-state-version-id"`
+	NameIndex                string `jsonapi:"attr,name-index"`
```
Comment (brandonc) [inline workspace_resources.go:46]: Can you make this field type a `*string` to reflect the fact that it can be `null`?

**Review by brandonc — COMMENTED** [review]
This looks great! Thanks for making the contribution. I only found one very small tweak. What's really surprising is that the README says that the resource is available when it was, in fact, not.

**Comment by brandonc** [comment]
@stefan-kiss We're actually prepping a release right now so I'm going to bring your commits into the repo and add the quick change I recommended.

**Comment by stefan-kiss** [comment]
> @stefan-kiss We're actually prepping a release right now so I'm going to bring your commits into the repo and add the quick change I recommended.

sorry, i missed the earlier comments but yes, it's perfectly fine. didnt realise it could be null :) anyway glad it was merged & thank you!

**Pull request merged**